lib/datasource/index.ts

```diff
@@ -49,12 +145,33 @@ async function fetchReleases(
   }
   const datasource = await load(datasourceName);
   const registryUrls = resolveRegistryUrls(datasource, config.registryUrls);
-  let dep = await datasource.getReleases({
-    ...config,
-    registryUrls,
-  });
-  if (!(dep && dep.releases.length)) {
-    dep = null;
+  let dep: ReleaseResult;
+  if (datasource.registryStrategy) {
+    if (!registryUrls.length) {
+      logger.warn(
+        { datasource: datasourceName, depName: config.depName },
+        'Missing registryUrls for registryStrategy'
+      );
+      return null;
+    }
+    if (datasource.registryStrategy === 'first') {
+      dep = await firstRegistry(config, datasource, registryUrls);
+    } else if (datasource.registryStrategy === 'hunt') {
+      dep = await huntRegistries(config, datasource, registryUrls);
+    } else if (datasource.registryStrategy === 'merge') {
+      dep = await mergeRegistries(config, datasource, registryUrls);
+    }
+  } else {
+    dep = await datasource.getReleases({
+      ...config,
+      registryUrls,
+    });
+  }
+  if (!dep) {
+    return null;
+  }
+  if (!dep.releases?.length) {
```

Comment:
```suggestion
  if (!dep?.releases?.length) {
```

Remove the check above